File: Design-Patterns/Builder-Design-Pattern/java/src/builderdesignpattern/GeneralDentist.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package builderdesignpattern;

import dentist.Address;
import dentist.Education;
import java.util.ArrayList;

import dentist.PersonalInformation;

/**
 *
 * @author yousef
 */
public class GeneralDentist implements DentistBuilder {
    
    private Dentist dentist;
    private PersonalInformation personalInformation;
    private ArrayList<Education> educations;
    private ArrayList<Address> addresses;

    public GeneralDentist(PersonalInformation personalInformation, ArrayList<Education> educations, ArrayList<Address> addresses) {
        this.dentist = new Dentist();
        this.personalInformation = personalInformation;
        this.educations = educations;
        this.addresses = addresses;
    }
    
    @Override
    public void buildPersonalInformation() {
        dentist.setPersonalInformation(personalInformation);
    }

    @Override
    public void buildEducation() {
        dentist.setEducation(educations);
    }

    @Override
    public void buildAddress() {
        dentist.setAddress(addresses);
    }

    @Override
    public Dentist getDentist() {
        return dentist;
    }
}

File: Design-Patterns/Builder-Design-Pattern/php/DentistBuilder/GeneralDentist.php
<?php
 

class GeneralDentist implements DentistBuilder {
    
    private $dentist;

    public function __construct() {
        $this->dentist = new Dentist();
    }
    
    public function buildPersonalInformation() {
        $this->dentist.setPersonalInformation();
    }

    public function buildEducation() {
        $this->dentist.setEducation();
    }

    public function buildAddress() {
        $this->dentist.setAddress();
    }

    public function buildSpecializationArea() {
        $this->dentist.setSpecializationArea();
    }

    public function getDentist() {
        return $this->dentist;
    }
}
